src/main/webapp/creatives/maps/simid_map_creative.test.js

```diff
@@ -181,36 +192,55 @@ test('testing that rejection for JSON parsing errors is working', () => {
         type: "SIMID:Player:init",
     };
     testMap.onInit(eventData);
-    
-    const instance = SimidProtocol.mock.instances[0];
-    const rejectMessageObject = instance.reject.mock.calls[0][1];
+
+    expect(mockReject.mock.calls.length).toBe(1);
+    const rejectMessageObject = mockReject.mock.calls[0][1];
     expect(rejectMessageObject.message).toBe("Invalid JSON input for ad parameters");
 });
 
-test('instance of map is instantiated on button click', () => {
+test('instance of map is instantiated if ad paused', async () => {
     const eventData = createInitData();
     testMap.onInit(eventData);
     testMap.onStart(startData);
+
     const findNearestButton = document.getElementById('findNearest');
     findNearestButton.dispatchEvent(new Event('click'));
+
+    const flushPromises = () => new Promise(setImmediate);
+    await flushPromises();
+
+    const returnButton = document.getElementById("returnToAd");
+    expect(returnButton.textContent).toBe('Return To Ad');
+    const skipButton = document.getElementById("skipAd");
+    expect(skipButton.textContent).toBe('Skip Ad');
     expect(window.google.maps.Map.mock.instances.length).toBe(1);
 });
 
-test('marker is added to map when map loads', () => {
+test('marker is added to map when map loads', async () => {
     const eventData = createInitData();
     testMap.onInit(eventData);
     testMap.onStart(startData);
+
     const findNearestButton = document.getElementById('findNearest');
     findNearestButton.dispatchEvent(new Event('click'));
+
+    const flushPromises = () => new Promise(setImmediate);
```

Comment:
Can make this a util function in the global scope

Reply:
done